monai/engines/trainer.py

```diff
@@ -153,14 +157,20 @@ def _iteration(self, engine: Engine, batchdata: Dict[str, torch.Tensor]):
         if self.amp and self.scaler is not None:
             with torch.cuda.amp.autocast():
                 predictions = self.inferer(inputs, self.network, *args, **kwargs)
```

Comment:
this won't support all the use cases, for example, we might need an accumulated trainer like this: https://gist.github.com/thomwolf/ac7a7da6b1888c2eeac8ac8b9b05d3d3#file-gradient_accumulation-py

Reply:
Hi @wyli ,

You are right, we don't suppose to use `Handler` for any cases, just want to cover most common cases and recommend users to use below 3 levels API:
```
1. Handlers based on rich Events -- try to cover most of the common cases, that's what this PR want to do.
2. iteration_update -- write own arbitrary iteration logic.
3. Extend SupervisedTrainer -- Do the most advanced features that current SupervisedTrainer can't satisfy.
```
For your example, user can write their own `optimizer` or `iteration_update`.

Thanks.